op-service/clock/clock.go

```diff
@@ -92,6 +97,14 @@ func (s systemClock) After(d time.Duration) <-chan time.Time {
 	return time.After(d)
 }
 
+func (s systemClock) Add(d time.Duration) time.Time {
+	now := s.Now()
+	if d < 0 && math.Abs(d.Seconds()) > float64(now.Unix()) {
+		return time.Unix(0, 0)
+	}
```

Comment:
This seems unexpected and potentially dangerous.  I'm not sure what the right thing to do is when time underflows but seems like just capping at 0 may not always be the right thing.